WebService/v2/Models/DeviceModelApiModel/DeviceModelSimulationScript.cs

```diff
@@ -0,0 +1,148 @@
+
+        [JsonProperty(PropertyName = "Path")]
+        public string Path { get; set; }
+
+        [JsonProperty(PropertyName = "Params", NullValueHandling = NullValueHandling.Ignore)]
+        public object Params { get; set; }
+
+        public DeviceModelSimulationScript()
+        {
+            this.Type = string.Empty;
+            this.Path = string.Empty;
+            this.Params = null;
+        }
+
+        // Map API model to service model
+        public Script ToServiceModel()
+        {
+            if (this.IsEmpty()) return null;
+
+            return new Script
+            {
+                Type = !string.IsNullOrEmpty(this.Type) ? this.Type : null,
+                Path = !string.IsNullOrEmpty(this.Path) ? this.Path : null,
+                Params = this.Params
+            };
+        }
+
+        // Map service model to API model
+        public static DeviceModelSimulationScript FromServiceModel(Script value)
+        {
+            if (value == null) return null;
+
+            return new DeviceModelSimulationScript
+            {
+                Type = string.IsNullOrEmpty(value.Type) ? null : value.Type,
+                Path = string.IsNullOrEmpty(value.Path) ? null : value.Path,
+                Params = value.Params
+            };
+        }
+
+        public bool IsEmpty()
+        {
+            return string.IsNullOrEmpty(this.Type)
+                   && string.IsNullOrEmpty(this.Path)
+                   && this.Params == null;
+        }
+
+        public void ValidateInputRequest(ILogger log)
+        {
+            const string INTERNAL = "internal";
+            const string NO_TYPE = "Script must contains a valid type";
+            const string NO_PATH = "Script must contains a valid path";
+
+            if (string.IsNullOrEmpty(this.Type))
+            {
+                log.Error(NO_TYPE, () => new { Script = this });
+                throw new BadRequestException(NO_TYPE);
+            }
+
+            if (string.IsNullOrEmpty(this.Path))
+            {
+                log.Error(NO_PATH, () => new { Script = this });
+                throw new BadRequestException(NO_PATH);
+            }
+
+            if (this.Type == INTERNAL)
+            {
+                this.ValidateParams(log);
+            }
+        }
+
+        private void ValidateParams(ILogger log)
+        {
+            if (this.Params == null)
+            {
+                this.ThrowInvalidParams(log);
+            }
+            
+            var rootObject = JObject.Parse(this.Params.ToString());
+            var values = rootObject.First?.First;
+
+            foreach (var token in rootObject)
+            {
+                var value = token.Value;
+                if (value == null)
+                {
+                    this.ThrowInvalidParams(log);
+                }
+
+                this.CheckProperties(log, value);
+            }
+        }
+
+        private void CheckProperties(ILogger log, JToken propValue)
```

Comment:
for the time being comment this method out, so we leave room for internal scripts with different params. i.e. Min/Max might not be needed by future scripts

Reply:
add a TODO: the actual function (e.g. the "math.random.withinrange") should provide a validation method

Reply:
Done.